src/models/Projection.ts

```diff
@@ -92,22 +123,77 @@ export class Projection {
     }));
   }
 
-  getDataset(columnName: string, customLabel?: string): ProjectionDataset {
+  getDataset(dataset: DatasetId, customLabel?: string): ProjectionDataset {
     return {
       label: customLabel ? customLabel : this.label,
-      data: this.getColumn(columnName),
+      data: this.getColumn(dataset),
     };
   }
 
+  private calcRtRange(timeseries: Timeseries): Array<RtRange | null> {
+    return timeseries.map(row => {
+      // TODO(michael): Why are the types wrong? I think
+      // json-schema-to-typescript may be broken. :-(
+      const rt = (row.Rt as any) as number;
+      const ci = (row.RtCI90 as any) as number;
+      if (rt) {
+        return {
+          rt: round(rt, 2),
+          low: round(rt - ci, 2),
+          high: round(rt + ci, 2),
+        };
+      } else {
+        return null;
+      }
+    });
+  }
+
+  private calcTestPositiveRate(timeseries: Timeseries): Array<number | null> {
+    const dailyPositives = this.deltasFromCumulatives(
+      timeseries.map(row => row.cumulativePositiveTests),
+    );
+    const dailyNegatives = this.deltasFromCumulatives(
+      timeseries.map(row => row.cumulativeNegativeTests),
+    );
+
+    return dailyPositives.map((positive, idx) => {
+      const negative = dailyNegatives[idx];
+      const total = positive + negative;
+      return total > 0 ? round(positive / total, 3) : null;
+    });
+  }
+
+  // Given a series of cumulative values, convert to a series of deltas.
+  // Treats null values as 0s.
+  private deltasFromCumulatives(cumulatives: Array<number | null>) {
```

Comment:
Will this end up with a large first delta? Maybe we chop it off - thought would have to figure out how to keep alignment with dates?

Reply:
Yeah...  I was debating a little about that but decided to keep it simple for now since it doesn't really matter for test positive rate where we use this (since we calculate a percentage, the large initial delta is fine).

But this will break down as soon as we decide to chart daily cases or anything like that, so I went ahead and reworked it.  While I was at it, I added smarter null-handling just to make it more robust for when we do start throwing more data at it.